### ISSUE TITLE
Count article views for 1 day 1 week etc

### ISSUE THREAD

**Issue by campgurus** [BOWiki/BOW#677]
currently, we can count only the total number of visits where the landing page corresponded to a specific article.

**Comment by campgurus** [comment]
I do not think this is working as is. Need to test on staging for a while. rolling back.

**Comment by trystant** [comment]
I think it would help to detail what is missing from the goal.

What other metrics do we want to have? What is the current error message and how are things off?

**Comment by trystant** [comment]
In the [index.html.erb](https://github.com/BOWiki/BOW/blob/master/app/views/analytics/index.html.erb) for Analytics, there are at least three lines with multiple compound scopes in the view:

104: ` <% Article.order(updated_at: :desc).first(10).each_with_index do |article, i| %>`
133: `<% Article.order(follows_count: :desc).first(10).each_with_index do |article, i| %>`
160: `<% Visit.where(started_at: 7.days.ago..Time.now).group(:landing_page).order('count_id DESC').limit(13).count(:id).each_with_index do |visit, i| %>`

Each of these should be  a custom scope that is associated with the model. The following lines should read as follows:

104: ` <% Article.recently_updated do |article, i| %>`
133: `<% Article.most_followed do |article, i| %>`
160: `<% Visit.most_in_last_week do |visit, i| %>`

and the logic in that index should be in those scopes.

**Comment by campgurus** [comment]
Ok this works.  When we were looking at views all time, I had removed the first element on the list, since it would always be the root page.  We currently look at the past 7 days, so that is no longer appropriate.

One remaining issue is that depending on how many non-article pages are included in the top viewed articles the length of the list changes.  That will be alleviated when we implement uniform fixed lengths for these lists and scrolling which is the subject of another issue.

Closing this.
